### PR TITLE
(cloudwatch) high resolution custom metrics

### PR DESCRIPTION
**Pull request by mtanda**
Support this.

https://aws.amazon.com/blogs/aws/new-high-resolution-custom-metrics-and-alarms-for-amazon-cloudwatch/

To support this feature, need to update aws-sdk-go, I update the sdk in this PR.

### CODE REVIEW

File: public/app/plugins/datasource/cloudwatch/datasource.js [hunk 1]
```diff
@@ -80,23 +80,28 @@ function (angular, _, moment, dateMath, kbn, templatingVariable, CloudWatchAnnot
...
-        periodUnit = period = 60 * 5;
-      } else if (now - start > (daySec * 63)) { // until 455 days ago
-        periodUnit = period = 60 * 60;
-      } else if (now - start > (daySec * 455)) { // over 455 days, should return error, but try to long period
-        periodUnit = period = 60 * 60;
-      } else if (!target.period) {
-        period = (query.namespace === 'AWS/EC2') ? 300 : 60;
-      } else if (/^\d+$/.test(target.period)) {
-        period = parseInt(target.period, 10);
+      if (!target.period) {
```
Comment (torkelo) [inline public/app/plugins/datasource/cloudwatch/datasource.js]: maybe this period logic could be extracted to a seperate function, and given some unit test?

**Comment by mtanda** [comment]
Add test, and fix bug....

File: public/app/plugins/datasource/cloudwatch/datasource.js (hunk 1 above)
Comment (torkelo) [inline public/app/plugins/datasource/cloudwatch/datasource.js]: sorry, stupid comment, it was already in a separate function :) did not see it in the diff, though it was all in the query method :)

**Comment by mtanda** [comment]
Thanks!!

**Pull request merged**